sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -1504,6 +1504,22 @@ object GpuOverrides {
           GpuPythonUDF(a.name, a.func, a.dataType,
             childExprs.map(_.convertToGpu()),
             a.evalType, a.udfDeterministic, a.resultId)
+
+        override def tagExprForGpu(): Unit = {
+          // Allow array type data only in output for the top Python UDF. Because seems
+          // there is something wrong in writing array type data by cuDF arrow IPC writer.
+          wrapped.children
+            .filter(_.dataType.isInstanceOf[ArrayType])
+            .foreach(e => willNotWorkOnGpu(s"array type data($e) as the input of Python UDF is" +
+              " not supported on GPU"))
+        }
+
+        // Allow only one level array type data which has been verified now.
+        // Will update this after verifying the nested array type.
+        override def isSupportedType(t: DataType): Boolean = t match {
```

Comment:
nit: I think you can achieve the same thing with

```
GpuOverrides.isSupportedType(allowArray=true, allowNesting=false)
```

`allowNesting` is false by default so it is probably not needed.

Reply:
Oh ye, updated. Looks like i did not fully understand this API.